test/integration/spec/connect.js

```diff
@@ -103,6 +104,52 @@ describe('connect', function() {
     });
   });
 
+  describe('region selection ', async () => {
+    let token;
+    let cancelablePromise;
+
+    beforeEach(() => {
+      const identity = randomName();
+      token = getToken(identity);
+    });
+
+    context('with an invalid region', async () => {
```

Comment:
@makarandp0 You can simplify the cases here and re-use some of the test code:
```js
['without', 'global', 'specific', 'invalid'].forEach(regionType => {
  const regionOptions = {
    global: { region: 'gll' },
    invalid: { region: 'foo' },
    specific: { region: 'de1' },
    without: {}
  }[regionType];

  context(`when called ${regionType === 'without a' ? regionType : `with a(n) ${regionType}`} region`, () => {
    beforeEach(() => {
      cancelablePromise = connect(token, Object.assign({}, defaults, regionOptions, { tracks: [] }));
    });

    if (regionType === 'invalid') {
      it('should return a CancelablePromise that rejects with a SignalingConnectionError', async () => {
        ...
      });
    } else {
      it('should return a CancelablePromise that resolves with a Room', async () => {
        ...
      });
    }
  });
});
```

Reply:
Thanks for the suggestion, that does look more compact.